src/components/protection/AddProtectionSingle.vue

```diff
@@ -416,39 +415,27 @@ export default class AddProtectionSingle extends BaseComponent {
     });
   }
 
-  async loadAvailableSpace() {
-    this.loadingAvailableSpace = true;
-    this.amountToMakeSpace = await vxm.ethBancor.getAvailableSpace({
-      poolId: this.pool.id
-    });
-    this.loadingAvailableSpace = false;
-  }
-
-  async loadMaxStakes() {
-    if (this.loadingMaxStakes) return;
-    this.loadingMaxStakes = true;
-    try {
-      const result = await vxm.ethBancor.getMaxStakesView({
-        poolId: this.pool.id
-      });
-      result.forEach(async x => {
-        if (x.token === this.token.symbol) {
-          this.maxStakeAmount = x.amount;
-          this.maxStakeSymbol = x.token;
-        }
-      });
-    } catch (e) {
-      console.log(e);
-    } finally {
-      this.loadingMaxStakes = false;
-    }
-  }
+  async load() {
+    if (this.loading) return;
+    this.loading = true;
 
-  async fetchTknSpaceAvailable() {
-    const result = await vxm.ethBancor.getMaxStakesView({
+    const res = await vxm.ethBancor.getAvailableAndAmountToGetSpace({
       poolId: this.pool.id
     });
-    this.tknSpaceAvailableLTOne = new BigNumber(result[1].amount).lt(1);
+    this.tknSpaceAvailableLTOne = res[0] as boolean;
+    const availableSpace = res[1] as {
+      amount: string;
+      token: string;
+    }[];
```

Comment:
Because of the potential inconsistent return type of ```getAvailableAndAmountToGetSpace``` we're now force to re-write and assume them.

Let's return an object from this method instead of an array and ensure it's consistent. Then we don't have to perform write any ```as``` .